### PR TITLE
feat: Add scalar function humanize

### LINKED ISSUE #5035: Feature(function): add scalar function  humanize
**Summary**

Description for this feature.

## Context:
```
databend > select sum(length(stack_trace)), sum( length(session_settings) ), sum(length(query_text )) from system.query_log;
+--------------------------+-------------------------------+-------------------------+
| sum(length(stack_trace)) | sum(length(session_settings)) | sum(length(query_text)) |
+--------------------------+-------------------------------+-------------------------+
|                        0 |                          6375 |                     734 |
+--------------------------+-------------------------------+-------------------------+
```

We can add `humanize` function to display the size for better reading.

- Function:  `humanize`
- ReturnType: String 
- Example:

```
databend > select humanize(1024);

+------------+
|  humanize(1024) |
+------------+
|       1.00 KiB |

```
- Refer Docs:  https://github.com/datafuselabs/databend/blob/790ddcde7fefee3795103bd503f8df57796e1bb1/docs/doc/100-faq/40-how-to-write-scalar-functions.md
- Utils: https://github.com/datafuselabs/databend/blob/3a1a529c1d8839db84c62004b7223b1f7390ad77/common/io/src/utils.rs

### PR DESCRIPTION
**Pull request by cadl**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Add new scalar function `humanize_size` and  `humanize_number`.

Example:
```sql
mysql> SELECT HUMANIZE_NUMBER(11230000.0);
+---------------------------+
| HUMANIZE_NUMBER(11230000) |
+---------------------------+
| 11.23 million             |
+---------------------------+

mysql> SELECT HUMANIZE_SIZE(1024 * 1024);
+------------------------------+
| HUMANIZE_SIZE((1024 * 1024)) |
+------------------------------+
| 1.00 MiB                     |
+------------------------------+
1 row in set (0.03 sec)
```

## Changelog

- New Feature

## Related Issues

Fixes #5035

### CODE REVIEW

**Comment by cadl** [comment]
WIP on https://github.com/datafuselabs/databend/issues/5035#issuecomment-1111550853

File: tests/suites/0_stateless/02_function/02_0055_function_strings_humanize.sql [hunk 1]
```diff
@@ -0,0 +1,22 @@
+SELECT HUMANIZE_SIZE(1000);
```
Comment (sundy-li) [inline tests/suites/0_stateless/02_function/02_0055_function_strings_humanize.sql:1]: I think we should modify `let delimiter = 1000_f64;` into `let delimiter = 1024_f64;`.

So:

1. `HUMANIZE_SIZE(1000)` -> `1000.00 B`.
2. `HUMANIZE_SIZE(1024)` -> `1.00 KB`.
3. `HUMANIZE_SIZE(1229);` -> `1.20 KB`

Comment (sundy-li) [reply tests/suites/0_stateless/02_function/02_0055_function_strings_humanize.sql:1]: 1 kB is 1000 bytes (Decimali.e. 10^3 is kilo)
1 KB/KiB is 1024 bytes (Binary i.e. 2^10)

We use `KB` now.

Comment (cadl) [reply tests/suites/0_stateless/02_function/02_0055_function_strings_humanize.sql:1]: > 1 kB is 1000 bytes (Decimali.e. 10^3 is kilo) 1 KB/KiB is 1024 bytes (Binary i.e. 2^10)
> 
> We use `KB` now.

kB/KB/MB always leads to misunderstandings, maybe we should just use the `KiB`.

https://en.wikipedia.org/wiki/Byte#Multiple-byte_units

Comment (sundy-li) [reply tests/suites/0_stateless/02_function/02_0055_function_strings_humanize.sql:1]: > maybe we should just use the KiB

Ok !

Comment (cadl) [reply tests/suites/0_stateless/02_function/02_0055_function_strings_humanize.sql]: @sundy-li updated. PTAL 🙏

**Review by sundy-li — APPROVED** [review]
Impressive！

**Review by BohuTANG — APPROVED** [review]
💯

**Comment by BohuTANG** [comment]
@mergify update

**Pull request merged**